scripts/mod_loader/modui/weapon_deck_selector.lua

```diff
@@ -0,0 +1,470 @@
+		else
+			-- load preset from config
+			local preset = {}
+			sdlext.config(modApi:getCurrentModcontentPath(), function(config)
+				if config.shopWeaponPresets ~= nil then
+					preset = config.shopWeaponPresets[presetId] or {}
+				end
+			end)
+			enabled = function(id) return preset[id] or false end
+		end
+		-- update buttons based on the preset
+		for _, button in ipairs(buttons) do
+			button.checked = enabled(button.id) or false
+		end
+		return true
+	end
+
+	-- main UI logic
+	sdlext.showDialog(function(ui, quit)
+		ui.onDialogExit = onExit
+
+		-- main frame
+		local frametop = Ui()
+			:width(0.8):height(0.8)
+			:posCentered()
+			:caption(GetText("ModContent_Button_ConfigureWeaponDeck"))
+			:decorate({ DecoFrameHeader(), DecoFrame() })
+			:addTo(ui)
+		-- scrollable content
+		local scrollArea = UiScrollArea()
+			:width(1):height(1)
+			:addTo(frametop)
+		-- define the window size to fit as many weapons as possible, comes out to about 5
+		local weaponsPerRow = math.floor(ui.w * frametop.wPercent / CELL_WIDTH)
+		frametop
+			:width((weaponsPerRow * CELL_WIDTH + (2 * PADDING)) / ui.w)
+			:posCentered()
+		ui:relayout()
+
+		-- add button area on the bottom
+		local line = Ui()
+      :width(1):heightpx(frametop.decorations[1].bordersize)
+      :decorate({ DecoSolid(frametop.decorations[1].bordercolor) })
+      :addTo(frametop)
+		local buttonLayout = UiBoxLayout()
+      :hgap(20)
+      :padding(24)
+      :width(1)
+      :addTo(frametop)
+		buttonLayout:heightpx(BUTTON_HEIGHT + buttonLayout.padt + buttonLayout.padb)
+		ui:relayout()
+		scrollArea:heightpx(scrollArea.h - (buttonLayout.h + line.h))
+		line:pospx(0, scrollArea.y + scrollArea.h)
+		buttonLayout:pospx(0, line.y + line.h)
+
+		-------------
+		-- Buttons --
+		-------------
+		local enableSaveLoad
+
+		--- Button to enable all weapons
+		local size = weaponsPerRow > 6 and 1.5 or 1
+		local enableAllButton = Ui()
+			:widthpx(WEAPON_WIDTH * size):heightpx(BUTTON_HEIGHT)
+			:settooltip(GetText("ConfigureWeaponDeck_EnableAll_Tooltip"))
+			:decorate({
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_EnableAll_Title")),
+			})
+			:addTo(buttonLayout)
+		function enableAllButton.onclicked()
+			for _, button in ipairs(buttons) do
+				button.checked = true
+			end
+			enableSaveLoad(true)
+			return true
+		end
+		--- Button to disable all weapons
+		local disableAllButton = Ui()
+			:widthpx(WEAPON_WIDTH * size):heightpx(BUTTON_HEIGHT)
+			:settooltip(GetText("ConfigureWeaponDeck_DisableAll_Tooltip"))
+			:decorate({
+				DecoButton(),
+				DecoAlign(0, 2),
+				DecoText(GetText("ConfigureWeaponDeck_DisableAll_Title")),
+			})
+			:addTo(buttonLayout)
+		function disableAllButton.onclicked()
+			for _, button in ipairs(buttons) do
+				button.checked = false
+			end
+			enableSaveLoad(true)
+			return true
+		end
+
+		-- add spacer before preset buttons
+		-- width is crafted to right align the preset buttons
```

Comment:
Instead of creating a dummy spacer element, you should be able to simply set `alignH = "right"` on the button elements (`UiBoxLayout` feature)